cnxarchive/views.py

```diff
@@ -115,6 +115,32 @@ def is_latest(id, version):
     return get_latest_version(id) == version
 
 
+def get_state(cursor, id, version):
+    """Determine the state of the content."""
+    version_split = version.split(".")
+    args = [id]
+    args.extend(version_split)
+
+    sql_statement = """
+    SELECT ms.statename
+    FROM modules as m
+    JOIN modulestates as ms
+    ON m.stateid=ms.stateid
+    WHERE uuid=%s
+    AND major_version=%s
+    """
+    if len(version_split) == 2:
+        sql_statement += "AND minor_version=%s"
+    sql_statement += ";"
+
+    cursor.execute(sql_statement, args)
+    temp = cursor.fetchone()
+    if temp is None:
+        return u'No Stateid in Modules table'
+    else:
+        return temp[0]
```

Comment:
While commendable to consider error cases, the extra guard code for no return here is not needed - `modules.stateid` has a foreign key relationship with the `modulestates` table, so it's not possible to put a value in that field that does not have a corresponding row in the other table.